core/src/main/scala/kafka/log/Log.scala

```diff
@@ -79,6 +78,17 @@ case class LogAppendInfo(var firstOffset: Long,
  *
  * @param dir The directory in which log segments are created.
  * @param config The log configuration settings
+ * @param logStartOffset The smallest offset is allowed to be exposed to kafka client.
+ *                       The logStartOffset can be updated by :
+ *                       - user's PurgeRequest
+ *                       - broker's log retention
+ *                       - broker's log truncation
```

Comment:
If it is not an expected behavior, we should throw some exception instead of allowing the LW to decrease, right?